libraries/cygwin_helpers.rb

```diff
@@ -1,7 +1,22 @@
 module Cygwin
 	module Helpers
+        require 'uri'
+        include Chef::Mixin::ShellOut
+
         def proxy_command
-            Chef::Config['http_proxy'].nil? ? "" : "--proxy #{Chef::Config['http_proxy']}"
+        	proxy = Chef::Config['http_proxy']
+        	return "" if proxy.nil?
+
+        	# The configured proxy often is in the form of a uri, whereas setup.exe
+        	# expects --proxy host:port
+        	proxy_uri = URI(proxy)
+        	proxy = "#{proxy_uri.host}:#{proxy_uri.port}" unless proxy_uri.host.nil?
+        	return "--proxy #{proxy}"
+        end
+
+        def is_installed?(cygpath, package)
+            cmd = shell_out!("cygcheck -c #{package}", {:cwd => "#{cygpath}/bin"})
+            !!(cmd.stdout =~ /\WOK\r\n$/)
```

Comment:
apparently =~ returns an integer (index where the match occurred) or nil when there was no match. It's OK to omit conversion to boolean, but I kept it in, just to avoid confusion when someone examines .exists and sees it as nil or some meaningless integer...